hw_config.h

```diff
@@ -781,6 +781,50 @@
 # define BOARD_USART_PIN_CLOCK_BIT      RCC_AHB1ENR_GPIOBEN
 # define SERIAL_BREAK_DETECT_DISABLED   1
 
+/****************************************************************************
+ * TARGET_HW_MATEKF405_WING
+ ****************************************************************************/
+
+#elif  defined(TARGET_HW_MATEKF405_WING)
+
+# define APP_LOAD_ADDRESS               0x08018000
+# define BOOTLOADER_DELAY               5000
+# define INTERFACE_USB                  1
+# define INTERFACE_USART                1
+# define USBDEVICESTRING                "Matek F405-WING"
+# define USBPRODUCTID                   0x0011
+# define BOOT_DELAY_ADDRESS             0x000001a0
+
+# define BOARD_TYPE                     127
+# define _FLASH_KBYTES                  (*(uint16_t *)0x1fff7a22)
+# define BOARD_FLASH_SECTORS            ((_FLASH_KBYTES == 0x400) ? 11 : 23)
+# define BOARD_FLASH_SIZE               (1024 * 1024)
+# define APP_RESERVATION_SIZE           (2 * 32 * 1024) /* 2 32 Kib Sectors */
```

Comment:
On F4, these are 16 KB sectors: https://github.com/PX4/Bootloader/blob/master/main_f4.c#L43.
You will also need to define `BOARD_FIRST_FLASH_SECTOR_TO_ERASE`.
I suggest to use only 1 sector so you'll have a bit more space for the FW.

Reply:
Will that still work for compatibility with Arduplane, similar to what you did in the KakuteF7?

Reply:
If this is a STM32F407/5? I would use the TARGET_HW_TAP_V1 as a reference for the memory map. IIRCC The FLASH based param require 2 sectors to work.

Reply:
One is enough. I did that on the omnibus.
Ardupilot is using 3 sectors.